base_bank_branch/views/res_bank_views.xml

```diff
@@ -0,0 +1,93 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<odoo>
+
+    <record id="view_res_bank_tree_inherit" model="ir.ui.view">
+        <field name="name">res.bank.tree.branch</field>
+        <field name="model">res.bank</field>
+        <field name="inherit_id" ref="base.view_res_bank_tree"/>
+        <field name="arch" type="xml">
+            <field name="country" position="after">
+                <field name="is_branch"/>
+                <field name="parent_id" invisible="1"/>
+            </field>
+        </field>
+    </record>
+
+    <record id="view_res_bank_form_inherit" model="ir.ui.view">
+        <field name="name">res.bank.form.branch</field>
+        <field name="model">res.bank</field>
+        <field name="inherit_id" ref="base.view_res_bank_form"/>
+        <field name="arch" type="xml">
+            <group name="bank_details" position="before">
+                <group
+                    name="branch_option"
+                    col="4"
+                    attrs="{'invisible': [('has_childs', '=', True)]}"
+                >
+                    <field name="is_branch"/>
```

Comment:
It looks strange, that `name` is not the first field in the form.

Reply:
Strange? partner form also has Is Individual/ Is Company at the top instead of name.